CRM/Event/Form/Registration.php

```diff
@@ -857,9 +840,7 @@ public static function addParticipant(&$form, $contactID) {
       'status_id' => CRM_Utils_Array::value('participant_status',
         $params, 1
       ),
-      'role_id' => CRM_Utils_Array::value('participant_role_id',
-        $params, $roleID
-      ),
+      'role_id' => self::getDefaultRoleID($params),
```

Comment:
How about 
```
'role_id' => self::getDefaultRoleID(
  CRM_Utils_Array::value('participant_role_id', $params)
),

```
`private static function getDefaultRoleID($roleID) {`

Reply:
or how about: 
```php
...
'role_id' => CRM_Utils_Array::value(
   'participant_role_id',
    $params,
    civicrm_api3('OptionValue', 'getvalue', [
        'return' => "value",
        'option_group_id' => "participant_role",
        'options' => ['limit' => 1],
      ])
  ),
...
```
:)

Reply:
I agree with creating a new function for getDefaultRoleID() since it can be re-used elsewhere but i think we should not pass whole params array to that function

Reply:
I agree - the only reason params is being passed is to check whether we need to look up the default role id - I think that should be done outside the function ie 

$roleId = CRM_Utils_Array::value('participant_role_id', $params) ? : self::getDefaultRoleID();